src/frontend/components/DashboardVideoLiveControlPane/widgets/DashboardVideoLiveWidgetVisibilityAndInteraction/index.tsx

```diff
@@ -0,0 +1,154 @@
+    id: 'component.DashboardVideoLiveWidgetVisibilityAndInteraction.updateVideoFail',
+  },
+});
+
+interface DashboardVideoLiveWidgetVisibilityAndInteractionProps {
+  video: Video;
+}
+
+export const DashboardVideoLiveWidgetVisibilityAndInteraction = ({
+  video,
+}: DashboardVideoLiveWidgetVisibilityAndInteractionProps) => {
+  const intl = useIntl();
+  const [visibilityChecked, setVisibilityChecked] = useState(video.is_public);
+
+  const videoMutation = useUpdateVideo(video.id, {
+    onSuccess: () => {
+      toast.success(intl.formatMessage(messages.updateVideoSucces), {
+        position: 'bottom-center',
+      });
+    },
+    onError: (err, variables) => {
+      if ('is_public' in variables) {
+        setVisibilityChecked(video.is_public);
+      }
+      report(err);
+      toast.error(intl.formatMessage(messages.updateVideoFail), {
+        position: 'bottom-center',
+      });
+    },
+  });
+
+  const onToggleChange = () => {
+    setVisibilityChecked(!video.is_public);
+    videoMutation.mutate({
+      is_public: !video.is_public,
+    });
+  };
+
+  useEffect(() => {
+    if (video.is_public) {
+      const clipboard = new ClipboardJS('.copy');
+      clipboard.on('success', () => {
+        toast(intl.formatMessage(messages.copySuccess), {
+          icon: '📋',
+          position: 'bottom-center',
+        });
+      });
+
+      clipboard.on('error', (event) => {
+        toast.error(event.text, {
+          position: 'bottom-center',
+        });
+      });
+      return () => clipboard.destroy();
+    }
+  }, [video.is_public]);
+
+  const publicVideoUrl = window.location.origin
+    .concat('/videos/')
+    .concat(video.id);
+
+  return (
+    <DashboardVideoLiveWidgetTemplate
+      infoText={intl.formatMessage(messages.info)}
+      initialOpenValue
+      title={intl.formatMessage(messages.title)}
+    >
+      <Box direction="column" gap="small">
+        <DashboardVideoLiveWidgetToggleInput
+          disabled={video.is_recording}
+          checked={visibilityChecked}
+          onChange={onToggleChange}
+          label={intl.formatMessage(messages.publiclyAvailableLabel)}
+        />
+        <DashboardVideoLiveWidgetDashedBoxCustom>
+          <Text
+            color={video.is_public ? 'blue-active' : '#b4cff2'}
+            size="0.875rem"
+            style={{ fontFamily: 'Roboto-Medium' }}
+            truncate
+          >
+            {publicVideoUrl}
+          </Text>
+
+          <Button
+            a11yTitle={intl.formatMessage(messages.copyButtonTitle)}
+            className={'copy'}
+            data-clipboard-text={publicVideoUrl}
+            disabled={!video.is_public}
+            plain
+            style={{ display: 'flex', padding: 0 }}
+            title={intl.formatMessage(messages.copyButtonTitle)}
+          >
+            <CopySVG iconColor="blue-active" width="20px" height="25px" />
+          </Button>
+        </DashboardVideoLiveWidgetDashedBoxCustom>
```

Comment:
Shouldn't this be visible only when the video is public ?

Reply:
idk, there is no specifications about it. For now, when video is not public, the copy-to-clipboard button is disabled.

Reply:
done in last commit